src/app/modules/shared_models/section_model.nim

```diff
@@ -295,13 +293,146 @@ QtObject:
 
   proc editItem*(self: SectionModel, item: SectionItem) =
     let index = self.getItemIndex(item.id)
-    if (index == -1):
+    if index == -1:
+      return
+    
+    var roles: seq[int] = @[]
+
+    if self.items[index].name != item.name:
```

Comment:
I wonder if there is some kind of more automatic way to do this kind of repetitive stuff...

Reply:
After quick research I came up with sth like that:

```nim
macro updateRole(p, r) = quote do:
  if self.items[index].`p` != item.`p`:
    self.items[index].`p` = item.`p`
    roles.add(ModelRole.`r`.int)

updateRole name, Name
updateRole memberRole, MemberRole
updateRole isControlNode, IsControlNode
updateRole description, Description
# ...
```
It shortens the code and reduces chance for accidental copy-paste bugs. But it seems to be the area for some nim experts :)